extensions/connect-to-vpn/src/menu-bar.tsx

```diff
@@ -17,10 +17,10 @@ export default function Command() {
   const isChecking = useRef(false);
 
   useEffect(() => {
-    if (environment.launchType === LaunchType.Background) {
+    if (environment.launchType === LaunchType.Background && !isLoading) {
       startRefresh();
     }
-  }, [environment.launchType, favoriteServices, otherServices, invalidServices]);
+  }, [environment.launchType, favoriteServices.length, otherServices.length, invalidServices.length]);
```

Comment:
Why these changes?

If the list changes but keeps the same length, the effect wouldn't be re-triggered.
And shouldn't `isLoading` be added to the deps list?

Reply:
Agree,`isLoading` must be in deps list. Added

Also explained within code, why we need to check array's length instead of shallow array memory check.

```
   // Check services array's length because of state changing in `fetchServiceStatus` function
    // That function changes `networkServices` state and updates whole effect, if we shallow check array memory address
```